src/content/docs/en/guides/integrations-guide/markdoc.mdx

```diff
@@ -0,0 +1,357 @@
+import { getEntryBySlug } from 'astro:content';
+import Heading from '../components/Heading.astro';
+
+const entry = await getEntryBySlug('docs', 'why-markdoc');
+const { Content } = await entry.render();
+---
+
+<Content
+  components={{ Heading }}
+/>
+```
+
+Now, all Markdown headings will render with the `Heading.astro` component. This example uses a level 3 heading, automatically passing `level: 3` as the component prop:
+
+```md
+### I'm a level 3 heading!
+```
+
+📚 [Find all of Markdoc's built-in nodes and node attributes on their documentation.](https://markdoc.dev/docs/nodes#built-in-nodes)
+
+#### Use client-side UI components
+
+Today, the `components` prop does not support the `client:` directive for hydrating components. To embed client-side components, create a wrapper `.astro` file to import your component and apply a `client:` directive manually.
+
+This example wraps a `Aside.tsx` component with a `ClientAside.astro` wrapper:
+
+```astro
+---
+// src/components/ClientAside.astro
+import Aside from './Aside';
+---
+
+<Aside client:load />
+```
+
+This component [can be applied via the `components` prop](#render-markdoc-nodes--html-elements-as-astro-components):
+
+```astro
+---
+// src/pages/why-markdoc.astro
+import { getEntryBySlug } from 'astro:content';
+import ClientAside from '../components/ClientAside.astro';
+
+const entry = await getEntryBySlug('docs', 'why-markdoc');
+const { Content } = await entry.render();
+---
+
+<Content
+  components={{
+    Aside: ClientAside,
+  }}
+/>
+```
+
+### Markdoc config
+
+The Markdoc integration accepts [all Markdoc configuration options](https://markdoc.dev/docs/config), including [tags](https://markdoc.dev/docs/tags) and [functions](https://markdoc.dev/docs/functions).
+
+You can pass these options from the `markdoc()` integration in your `astro.config`. This example adds a global `getCountryEmoji` function:
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import markdoc from '@astrojs/markdoc';
+
+// https://astro.build/config
+export default defineConfig({
+  integrations: [
+    markdoc({
+      functions: {
+        getCountryEmoji: {
+          transform(parameters) {
+            const [country] = Object.values(parameters);
+            const countryToEmojiMap = {
+              japan: '🇯🇵',
+              spain: '🇪🇸',
+              france: '🇫🇷',
+            }
+            return countryToEmojiMap[country] ?? '🏳'
+          },
+        },
+      },
+    }),
+  ],
+});
+```
+
+Now, you can call this function from any Markdoc content entry:
+
+```md
+¡Hola {% getCountryEmoji("spain") %}!
+```
+
+:::note
+These options will be applied during [the Markdoc "transform" phase](https://markdoc.dev/docs/render#transform). This is run **at build time** (rather than server request time) both for static and SSR Astro projects. If you need to define configuration at runtime (ex. SSR variables), [see the next section](#Define-markdoc-configuration-at-runtime).
```

Comment:
Nice try, Ben! Should be this I think:

```suggestion
These options will be applied during [the Markdoc "transform" phase](https://markdoc.dev/docs/render#transform). This is run **at build time** (rather than server request time) both for static and SSR Astro projects. If you need to define configuration at runtime (ex. SSR variables), [see the next section](#define-markdoc-configuration-at-runtime).
```

It can be helpful to check [the error message CI prints](https://github.com/withastro/docs/actions/runs/4359106000/jobs/7620510189#step:4:2654) as it lists out the available fragment links on the page.